packages/calcite-components/src/components/action-menu/action-menu.tsx

```diff
@@ -435,15 +435,25 @@ export class ActionMenu extends LitElement {
     this.open = value;
   }
 
-  private handlePopoverOpen(): void {
+  private handlePopoverOpen(event: CustomEvent<void>): void {
+    event.stopPropagation();
     this.open = true;
     this.setFocus();
   }
 
-  private handlePopoverClose(): void {
+  private handlePopoverClose(event: CustomEvent<void>): void {
+    event.stopPropagation();
     this.open = false;
   }
 
+  private handlePopoverBeforeClose(event: CustomEvent<void>): void {
+    event.stopPropagation();
```

Comment:
We only stop propagation of handled events. Do these also need to be prevented?

Reply:
These are handled in the time-picker. Maybe we need all the open/close events on action-menu? I can leave this one out and open a follow up issue to get action-menu consistent with other open close component events?

https://github.com/Esri/calcite-design-system/blob/dev/packages/calcite-components/src/components/input-time-picker/input-time-picker.tsx#L556

Reply:
The reason this is beneficial is because if someone is listening to `onCalciteBeforeClose` on a popover that contains an action-menu inside we don't want this to bubble up.